src/beamlime/applications/_nexus_helpers.py

```diff
@@ -243,50 +229,51 @@ def _merge_f144(group: NexusGroup, data: DeserializedMessage) -> None:
 
 
 def _find_tdct(
-    structure: NexusStructure, data: DeserializedMessage
-) -> Iterable[NexusPath]:
-    name = data["name"]
-    for path, node in iter_nexus_structure(structure):
-        if node.get("name") == name:
-            for _, child in iter_nexus_structure(node):
-                if child.get("module") == "tdct":
-                    yield (*path, 'top_dead_center')
+    modules: ModuleRegistry, data: DeserializedMessage
+) -> Generator[tuple[str, NexusGroup]]:
+    raise NotImplementedError("TDCT is not supported yet.")
 
 
 def _initialize_tdct(
-    structure: NexusStructure, path: NexusPath, data: DeserializedMessage
-) -> NexusDataset:
-    return create_dataset(
-        name="top_dead_center",
-        dtype="uint64",
-        initial_values=np.asarray([], dtype="uint64"),
-        unit="ns",
-    )
+    parent: NexusGroup, data: DeserializedMessage, module: NexusStreamedModule
+) -> NexusGroup:
+    group = parent.copy()
+    dtype = module["config"].get("dtype", "uint64")
+    group['children'] = [
+        create_dataset(
+            name=data["name"],
+            dtype=dtype,
+            initial_values=np.asarray([], dtype=dtype),
+            unit=module["config"].get("unit", "ns"),
+        )
+    ]
+    return group
```

Comment:
I'm pretty sure that `tdct` is only a dataset and not a group (unlike `ev44` and `f144`). That's why it was implemented the way it was.

See https://manual.nexusformat.org/classes/base_classes/NXdisk_chopper.html